server.js

```diff
@@ -2,153 +2,22 @@
-      federated: '',
-      returnTo: 'http://localhost:4000/logout/callback',
-    },
-  },
-  (profile, done) => {
-    // TODO: This will need to be changed when we switch to ADFS
-    const email =
-      profile[
-        'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'
-      ];
-    db.user
-      .findOne({
-        where: { email: { [Op.like]: email } },
-      })
-      .then((matchingUser) => {
-        // Save the nameId and nameIDFormat for logout
-        // Note: the only field from the user object we need is the email
-        const returnVal = {
-          userEmail: matchingUser.email,
-          saml: {
-            nameID: profile.nameID,
-            nameIDFormat: profile.nameIDFormat,
-          },
-        };
-        done(
-          returnVal.userEmail
-            ? null
-            : new Error('Email is not registered: ' + email),
-          returnVal
-        );
-      });
-  }
-);
-
-passport.use(samlStrategy);
-
-/** This saves user.email in the session, under req.session.passport.user,
- *    which is later used by deserializeUser to find the user object and attach
- *    it to req.user. After this operation, req.session.passport.user = user.email */
-
-passport.serializeUser(({ userEmail }, done) => {
-  done(null, userEmail);
-});
-
-/** This is called with the email that was saved in serializeUser as the
- *     first argument to this function. (i.e. email = user.email). We would then
- *     use the user's email to get the actual user object (using db.user.findOne(...))
- *     Afterwards, the user object is attached to req.user */
-passport.deserializeUser(async (email, done) => {
-  done(
-    null,
-    await db.user.findOne({
-      where: { email: { [Op.like]: email } },
-    })
-  );
-});
-
-const app = express();
-
-app.use(
-  session({
-    secret: SESSION_SECRET,
-    resave: false,
-    saveUninitialized: false,
-    // TODO: Handle login cookie expiring
-    // cookie: { maxAge: 1000 },
-  })
-);
-
-app.use(passport.initialize({}));
-app.use(passport.session({}));
-
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(bodyParser.json());
-
-app.get('/login', passport.authenticate('saml'));
-
-// TODO: Redirect to frontend auth landing page
-app.post('/login/callback', passport.authenticate('saml'), (req, res) => {
-  if (!req.user) {
-    res.send('Login unsuccessful');
-  } else {
-    res.send('Login successful!');
-  }
-});
-
-app.get('/logout', (req, res) => {
-  // TODO: Redirect to proper unauth screen upon logout
-  if (req.user == null) {
-    res.send('No logged in user!');
-    return;
-  }
-  samlStrategy.logout(req, function (err, uri) {
-    // Clear user login session on our end
-    res.clearCookie('connect.sid');
-    // Redirect to logout callback URL
-    // Note: this uri is specified in the `returnTo` field of `samlStrategy`'s `additionalLogoutParams`
-    return res.redirect(uri);
+if (require.main === module) {
```

Comment:
just to make sure I'm understanding this right, `require.main` corresponds to the file that we call with `node file.js`, where `require.main` would equal `file.js`, correct?

Reply:
i.e. it's the "entry point" into the app, the file we call with the command `node`?

Reply:
if so, how come this if statement is needed? If it's the case that `require.main !== module`, i.e. we're calling a DIFFERENT file, `server.js` wouldn't be called at all right since it isn't imported anywhere?

Reply:
i believe we want to make sure that the server is only initialized if we explicitly call `node server.js` as the entrypoint so we have full control over how it runs and what settings/environment it's running in